model/qti/CustomInteractionAsset/Extractor/Api/AssetExtractorInterface.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2021 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoQtiItem\model\qti\CustomInteractionAsset\Extractor\Api;
+
+use oat\taoQtiItem\model\qti\interaction\CustomInteraction;
+
+/**
+ * @author Kiryl Poyu <kyril.poyu@taotesting.com>
+ */
+interface AssetExtractorInterface
+{
+    /**
+     * @return array<string>
+     */
+    public function extract(CustomInteraction $interaction): array;
```

Comment:
Just a suggestion, WDYT of returning an `iterable` type here?
That will allow you to replace the implementation with a generator, instead of building the whole array of assets.

Reply:
agree with you